### PR TITLE
feat: setup instructor dashboard filter

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR setup the instructor dashboard filter integration:

image

## Testing instructions

- Install `tutor-contrib-aspects` and initialize the environment and checkout this branch https://github.com/openedx/tutor-contrib-aspects/pull/463
- Install this plugin
- Setup the following tutor inline plugin:
```yaml
name: settings
version: 0.1.0
patches:
      OPEN_EDX_FILTERS_CONFIG = {
        "org.openedx.learning.instructor.dashboard.render.started.v1": {
          "fail_silently": False,
          "pipeline": [
            "platform_plugin_superset.extensions.filters.AddSupersetTab",
          ]
        },
      }
```
- Initialize your environment via `tutor dev|local do init -l aspects`
- Go to the instructor dashboard or go to this link: http://local.overhang.io:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-superset

**Merge checklist:**
Check off if complete *or* not applicable:
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [ ] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

### CODE REVIEW

**Comment by mariajgrimaldi** [comment]
Could you add the Athena team to the collaborators list? So we can tag them as reviewers

File: platform_plugin_superset/extensions/filters.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    """
+    Generate a Superset guest token for the user.
+    """
+    SUPERSET_CONFIG = getattr(settings, "SUPERSET_CONFIG", {})
+    superset_internal_host = SUPERSET_CONFIG.get("service_url", "http://superset:8088/")
+    superset_username = SUPERSET_CONFIG.get("username")
+    superset_password = SUPERSET_CONFIG.get("password")
+
+    INSTRUCTOR_DASHBOARD_CONFIG = getattr(settings, "SUPERSET_INSTRUCTOR_DASHBOARD", {})
+    dashboard_id = INSTRUCTOR_DASHBOARD_CONFIG.get("dashboard_uuid")
```
Comment (mariajgrimaldi) [inline platform_plugin_superset/extensions/filters.py]: We usually use uppercase constants when defining them on top of the file so I don't think is needed here.

File: platform_plugin_superset/extensions/filters.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        password=superset_password,
+    )
+
+    course_run = course.children[0].course_key.run
+
+    data = {
+        "user": {
+            "username": user.username,
+            "first_name": "John",
+            "last_name": "Doe",
```
Comment (mariajgrimaldi) [inline platform_plugin_superset/extensions/filters.py:69]: why is this hardcoded? should we get the real users' name?

File: platform_plugin_superset/extensions/filters.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        "resources": [{"type": "dashboard", "id": dashboard_id}],
+        "rls": [
+            {"clause": f"org = '{(course.org)}'"},
+            {"clause": f"course_name = '{(course.display_name)}'"},
+            {"clause": f"course_run = '{(course_run)}'"},
+        ],
+    }
+
+    response = client.session.post(
+        url=superset_internal_host + "api/v1/security/guest_token/",
```
Comment (mariajgrimaldi) [inline platform_plugin_superset/extensions/filters.py]: can we use f strings consistently across the implementation?

File: platform_plugin_superset/extensions/filters.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    superset_password = SUPERSET_CONFIG.get("password")
+
+    INSTRUCTOR_DASHBOARD_CONFIG = getattr(settings, "SUPERSET_INSTRUCTOR_DASHBOARD", {})
+    dashboard_id = INSTRUCTOR_DASHBOARD_CONFIG.get("dashboard_uuid")
+
+    client = SupersetClient(
+        host=superset_internal_host,
+        username=superset_username,
+        password=superset_password,
+    )
```
Comment (mariajgrimaldi) [inline platform_plugin_superset/extensions/filters.py]: what would happen if this fails?

File: platform_plugin_superset/extensions/filters.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+            {"clause": f"course_name = '{(course.display_name)}'"},
+            {"clause": f"course_run = '{(course_run)}'"},
+        ],
+    }
+
+    response = client.session.post(
+        url=superset_internal_host + "api/v1/security/guest_token/",
+        json=data,
+        headers={"Content-Type": "application/json"},
+    )
```
Comment (mariajgrimaldi) [inline platform_plugin_superset/extensions/filters.py]: what would happen if the credentials are incorrect and this doesn't return 200 OK response?

File: platform_plugin_superset/extensions/filters.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    ):  # pylint: disable=arguments-differ, unused-argument
+        """Execute filter that modifies the instructor dashboard context.
+        Args:
+            context (dict): the context for the instructor dashboard.
+            _ (str): instructor dashboard template name.
+        """
+        course = context["course"]
+
+        user = get_current_user()
+        superset_token, dashboard_id = generate_guest_token(user, course)
```
Comment (mariajgrimaldi) [inline platform_plugin_superset/extensions/filters.py:105]: Same question about the errors here ^

**Review by mariajgrimaldi — CHANGES_REQUESTED** [review]

File: platform_plugin_superset/extensions/filters.py (hunk 2 above)
Comment (Ian2012) [reply platform_plugin_superset/extensions/filters.py:69]: It doesn't affect the user experience with the embedded dashboards, and Open edX doesn't have first and last name. What do you suggest  @mariajgrimaldi?

File: platform_plugin_superset/extensions/filters.py (hunk 4 above)
Comment (Ian2012) [reply platform_plugin_superset/extensions/filters.py]: And exception is raised, what should we do? Show the error message in the frontend? or Show an static error message with a suggestion to contact an Operator? @mariajgrimaldi

File: platform_plugin_superset/static/html/superset.html [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+                        filters: {
+                            expanded: false,
+                        },
+                        hideTab: true,
+                        hideChartControls: true,
+                        hideFilters: true,
+                    },
+                })
+                .then((dashboard) => {
+                    mountPoint = document.getElementById("superset-embedded-container");
```
Comment (johnvente) [inline platform_plugin_superset/static/html/superset.html:30]: I would pass this to a css file with this content and I'd delete this

```
#superset-embedded-container {
    height: 720px;
    width: 100%;
    display: block;
}

```

File: platform_plugin_superset/static/html/superset.html [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    <div>{{title}}</div>
+
+    <div id="superset-embedded-container" style="display: none">
+        <script type="text/javascript">
+            supersetEmbeddedSdk
+                .embedDashboard({
+                    id: "{{dashboard_id}}", // given by the Superset embedding UI
+                    supersetDomain: "{{superset_url}}", // your Superset instance
+                    mountPoint: document.getElementById("superset-embedded-container"), // any html element that can contain an iframe
+                    fetchGuestToken: () => {
```
Comment (johnvente) [inline platform_plugin_superset/static/html/superset.html]: fetchGuestToken: () => "{{superset_token}}",

**Comment by Ian2012** [comment]
@mariajgrimaldi @johnvente can you re review?

File: platform_plugin_superset/static/js/superset.js
Comment (johnvente) [inline platform_plugin_superset/static/js/superset.js:1]: Is this file necessary?

**Review by johnvente — APPROVED** [review]
LGTM! I've left you a last comment but is okay for me

File: platform_plugin_superset/static/js/superset.js
Comment (Ian2012) [reply platform_plugin_superset/static/js/superset.js:1]: I can refactor the js to be there, but it's optional

**Comment by Ian2012** [comment]
@mariajgrimaldi have in mind that an important part of the functionality of this PR will be refactored on #3

**Comment by Ian2012** [comment]
@mariajgrimaldi have in mind that an important part of the functionality of this PR will be refactored on #3

File: platform_plugin_superset/extensions/filters.py (hunk 2 above)
Comment (mariajgrimaldi) [reply platform_plugin_superset/extensions/filters.py:69]: does it fail if removed?

Comment (mariajgrimaldi) [reply platform_plugin_superset/extensions/filters.py:69]: The user has a full name, at least the could use it. If not, leave a comment explaining the dummy values. Thanks!

Comment (Ian2012) [reply platform_plugin_superset/extensions/filters.py:69]: It doesn't fail if removed

Comment (Ian2012) [reply platform_plugin_superset/extensions/filters.py:69]: fixed

File: platform_plugin_superset/extensions/filters.py [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+import pkg_resources
+from crum import get_current_user
+from django.conf import settings
+from django.template import Context, Template
+from openedx_filters import PipelineStep
+from supersetapiclient.client import SupersetClient
+from web_fragments.fragment import Fragment
+
+os.environ["OAUTHLIB_INSECURE_TRANSPORT"] = "1"
```
Comment (mariajgrimaldi) [inline platform_plugin_superset/extensions/filters.py:14]: why are we modifying this here? what does it affect?

**Review by mariajgrimaldi — APPROVED** [review]
Just a comment left to address.

File: platform_plugin_superset/extensions/filters.py (hunk 9 above)
Comment (Ian2012) [reply platform_plugin_superset/extensions/filters.py:14]: This is was extracted from here: https://github.com/eduNEXT/aspects-superset-aside/pull/2/files#diff-e5ef7509835b73d0294668644b6133c454ad335edf6a41742fe278b064a0236cR15-R18

At this point, I don't remember what it does

Comment (mariajgrimaldi) [reply platform_plugin_superset/extensions/filters.py:14]: what if we try it without it?

Comment (Ian2012) [reply platform_plugin_superset/extensions/filters.py:14]: Fixed in #3
